pkg/loop/internal/test/README.md

```diff
@@ -0,0 +1,65 @@
+This package contains test implementations for all the components that compose OCR2 and OCR3 LOOPPs.
+
+The directory layout is intended to mirror the natural interfaces composition of libocr:
+- ocr2: contains static test implementations for reuseable reporting plugin, factory, and plugin provider
+- ocr3: implements the ocr3 generic-based reporting plugin and reused common components from ocr2
+- resources: represents core-node provided resources used to in calls to core node api(s) eg pipeline_runner, keystore, etc
+- median: product specific test implementations required to fulfill the MedianProvider abstraction and median factory generator construction
+- mercury: product specific test implementation required to fulfill MercuryProvider and constructors
+
+Every test implementation follows the pattern wrapping an interface and provider one or more funcs to compare to another
+instance of the interface. Every package attempts to share exposed the bare minimum surface area to avoid entanglement with logically seperated tests and domains.
+
+In practice this is accomplished by exporting a static implementation and an interface that the implementation satisfies. The interface is used by other packages to declaration dependencies and the implementation is used in the test implementation of those other packaged
+
+Example
+
+```
+package x_test
+
+var FooImpl = staticFoo{
+    expectedStr = "a"
+    expectedInt = 1
+}
+
+type FooEvaluator interface {
```

Comment:
if i were more ambitious, i could likely turn this into a generic. that could be nice to enforce a bit more rigidity for new test implementations.

however, that is out of scope for now though as the real goal of this PR is to unblock CCIP. we can iterate on the implement details of the test suite after delivering OKR